articles/tutorial-qdk-grovers-search.md

```diff
@@ -11,49 +11,367 @@ title: Run Grover's search algorithm in Q# - Quantum Development Kit
+
+### Complete Grover's operation
+
+Now we are ready to write the full operation a Grover's search. It will have three inputs:
+
+- A qubit array `register : Qubit[]` that should be initialized in the all `Zero` state. This register will encode the tentative solution to the search problem. After the operation it will be measured.
+- An operation `phaseOracle : ((Qubit[]) => Unit is Adj` that represents the phase oracle for the Grover's task. This operation applies an unitary transformation over a generic qubit register.
+- An integer `iterations : Int` to represent the iterations of the algorithm.
+
+```qsharp
+operation RunGroversSearch(register : Qubit[], phaseOracle : ((Qubit[]) => Unit is Adj), iterations : Int) : Unit {
+    // Prepare register into uniform superposition.
+    ApplyToEach(H, register);
+    // Start Grover's loop.
+    for (_ in 1 .. iterations) {
+        // Apply phase oracle for the task.
+        phaseOracle(register);
+        // Apply Grover's diffusion operation.
+        ReflectAboutUniform(register);
+    }
+}
+```
+
+This code is generic - it can be used to solve any search problem. We pass the quantum oracle - the only operation that relies on the knowledge of the problem instance we want to solve - as a parameter to the search code.
+
+## Implement the oracle
+
+One of the key properties that makes Grover's algorithm faster is the ability of
+quantum computers of performing calculations not only on individual inputs but
+also on superpositions of inputs. We need to compute the function $f(x)$ that
+describes the instance of a search problem using only quantum operations. This
+way we can compute it over a superposition of inputs.
+
+Unfortunately there isn't an automatic way to translate classical functions to
+quantum operations. It's an open field of research in computer science called
+*reversible computing*.
+
+However, there are some guidelines that might help you to translate your
+function $f(x)$ into a quantum oracle:
+
+1. **Break down the classical function into small building blocks that are easy to implement.** For example, you can try
+   to decompose your function $f(x)$ into a series of arithmetic operations or boolean logic gates.
+1. **Use the higher-level building blocks of the Q# library operations to implement the intermediate operations.** For instance,
+   if you decomposed your function into a combination of simple arithmetic operations, you can use the [Numerics library](xref:microsoft.quantum.arithmetic) to implement the intermediate operations. 
+
+The following equivalence table might result you useful to implement boolean functions in Q#.
+
+| Classical logic gate | Q# operation        |
+|----------------|--------------------------|
+| $NOT$          | `X`                      |
+| $XOR$          | `CNOT`                   |
+| $AND$          | `CCNOT` with an auxiliary qubit|
+
+### Example: Quantum operation to check if a number is a divisor
+
+> [!IMPORTANT]
```

Comment:
What motivates factoring over something like the ISBN game used in MS Learn?

Reply:
This was written before that MS Learn module, but it was a pending PR in the previous module. But since this tutorial is focused on just implementing Grover's, I think a simple problem that doesn't require additional knowledge can be good for users to focus on Grover's, and not the problem.